login-utils/lslogins.c

```diff
@@ -818,7 +874,10 @@ static struct lslogins_user *get_user_info(struct lslogins_control *ctl, const c
 						sizeof(user_wtmp->ut_host),
 						sizeof(user_wtmp->ut_host) + 1);;
 			}  else
-				get_lastlog(ctl, user->uid, user->last_hostname, LASTLOG_HOST);
+#ifdef HAVE_LASTLOG2
+			        if (get_lastlog2(ctl, user->login, user->last_hostname, LASTLOG_HOST) < 0)
+#endif
+				        get_lastlog(ctl, user->uid, user->last_hostname, LASTLOG_HOST);
```

Comment:
The fallback logic could be put into a helper function instead of having the `#ifdef` three times.